connector/src/rest/connector/libs/apic/implementation.py

```diff
@@ -166,7 +166,7 @@ def decorated(self, *args, **kwargs):
             # Check if connected
             try:
                 self.disconnect()
-                self.connect()
+                self.connect(timeout)
```

Comment:
This will raise error, since timeout is not defined. I think you need to get the timeout from kwargs?

Reply:
Yes agreed be careful though because if the user doesnt explicitly pass 'timeout' arg it wont show up:
```
>>> def isconnected(func):
...   def decorated(*args, **kwargs):
...     print("THIS IS ISCONNECTED DECORATOR. *ARGS: {} **KWARGS: {}".format(args, kwargs))
...     return func(*args, **kwargs)
...   return decorated
...
>>>
>>>
>>> @isconnected
... def get(mandatory_arg, test_arg='123', timeout='30'):
...   print('THIS IS GET METHOD')
...
>>>
>>> get('test')
THIS IS ISCONNECTED DECORATOR. *ARGS: ('test',) **KWARGS: {}
THIS IS GET METHOD
>>>
>>>
>>> get('test', timeout='45')
THIS IS ISCONNECTED DECORATOR. *ARGS: ('test',) **KWARGS: {'timeout': '45'}
THIS IS GET METHOD